src/t8_cmesh.hxx

```diff
@@ -31,6 +31,59 @@
 #include <t8_cmesh/t8_cmesh_types.h>
 #include <t8_geometry/t8_geometry_handler.hxx>
 
+/**
+ * Compute the first element of a process in a partitioned mesh, via floor(process * global_num_elements / mpisize).
+ * Prevents overflow by splitting global_num_elements into two 32 bit integers.
+ * global_num_elements = a_0 * tau + a_1, tau = 2^32.
+ * 
+ * floor(a_0 * tau + a_1) * process / mpisize
+ * = floor (a_0 * tau * process / mpisize + a_1 * process / mpisize)
+ * (add the fractional part of the summands and take the floor) to split the computation into separate parts.
+ * = floor (a_0 * tau * process / mpisize) + floor (a_1 * process / mpisize) + floor ((a_o*tau*process) % mpisize + a_1*process % mpisize) / mpisize)
+ * the first term can computed as:
+ * floor (a_0 * tau * process / mpisize)
+ * = floor (floor (a_0 * tau / mpisize * process ) + (a_0 * tau % mpisize) * process / mpisize)
+ * We divide a_0 * tau by mpisize first to prevent an overflow. The second term corrects for the rounding error.
```

Comment:
Please update documentation to new computation

Reply:
Done